explorer/nodewatch/nodewatch/RoutesFacade.py

```diff
@@ -68,15 +69,35 @@ def html_nodes(self):
 			'explorer_endpoint': self.explorer_endpoint
 		})
 
-	def json_nodes(self, role, exact_match=False):
-		"""Returns all nodes with matching role."""
+	def json_nodes(self, **kwargs):
+		"""Returns all nodes with condition."""
 
-		def role_filter(descriptor):
-			return role == descriptor.roles if exact_match else role == (role & descriptor.roles)
+		role = kwargs.get('role')
+		exact_match = kwargs.get('exact_match')
 
-		return list(map(
+		limit = kwargs.get('limit')
+		only_ssl = kwargs.get('only_ssl')
+		order = kwargs.get('order')
+
+		def custom_filter(descriptor):
+			role_condition = True
+
+			if role is not None:
+				role_condition = role == descriptor.roles if exact_match else role == (role & descriptor.roles)
+
+			if only_ssl:
+				return role_condition and descriptor.is_ssl_enabled
```

Comment:
Should be able to do this ``return descriptor.is_ssl_enabled  if only_ssl else True``